ignoreTrailingComments option for max-len not being respected

<!--
This template is for bug reports. If you are reporting a bug, please continue on. If you are here for another reason, please see below:

1. To propose a new rule: http://eslint.org/docs/developer-guide/contributing/new-rules
2. To request a change: http://eslint.org/docs/developer-guide/contributing/changes
3. If you have any questions, please stop by our chatroom: https://gitter.im/eslint/eslint

Note that leaving sections blank will make it difficult for us to troubleshoot and we may have to close the issue.
-->

**What version of ESLint are you using?** 2.3.0

**What parser (default, Babel-ESLint, etc.) are you using?** default

**Please show your full configuration:**

```js
module.exports = {
  rules: {
    'max-len': [ 2, {
      code: 80,
      tabWidth: 2,
      ignoreUrls: true,
      ignoreTrailingComments: true
    }]
  }
}
```

**What did you do? Please include the actual source code causing the issue.**

```js
'use strict'

module.exports = {
  env: 'NODE_ENV' // This sets the config.get('env') to whatever NODE_ENV is set to
}
```

**What did you expect to happen?**

I expected there to not be an eslint error because I've set `ignoreTrailingComments` to `true`, which from the docs means:

> Only ignores comments that are trailing source.

**What actually happened? Please include the actual, raw output from ESLint.**

ESLint reported an error. It seems to not be respecting the `ignoreTrailingComments` rule

```
  4:1  error  Line 4 exceeds the maximum line length of 80  max-len
```

--- a/lib/rules/max-len.js
+++ b/lib/rules/max-len.js
@@ -118,7 +118,7 @@ module.exports = function(context) {
         // split (honors line-ending)
         var lines = context.getSourceLines(),
             // list of comments to ignore
-            comments = ignoreComments || maxCommentLength ? context.getAllComments() : [],
+            comments = ignoreComments || maxCommentLength || ignoreTrailingComments ? context.getAllComments() : [],
             // we iterate over comments in parallel with the lines
             commentsIndex = 0;
 

--- a/tests/lib/rules/max-len.js
+++ b/tests/lib/rules/max-len.js
@@ -76,6 +76,12 @@ ruleTester.run("max-len", rule, {
                 "// Full line comment\n" +
                 "someCode(); // With a long trailing comment.",
             options: [{code: 30, tabWidth: 4, comments: 20, ignoreTrailingComments: true}]
+        }, {
+            code: "var foo = module.exports = {}; // really long trailing comment",
+            options: [40, 4, {ignoreTrailingComments: true}]
+        }, {
+            code: "var foo = module.exports = {}; // really long trailing comment",
+            options: [40, 4, {ignoreComments: true, ignoreTrailingComments: false}]
         },
         // blank line
         ""
@@ -224,6 +230,17 @@ ruleTester.run("max-len", rule, {
             errors: [
                 {
                     message: "Line 1 exceeds the maximum line length of 20.",
+                    type: "Program",
+                    line: 1,
+                    column: 1
+                }
+            ]
+        }, {
+            code: "//This is very long comment with more than 40 characters which is invalid",
+            options: [40, 4, {ignoreTrailingComments: true }],
+            errors: [
+                {
+                    message: "Line 1 exceeds the maximum line length of 40.",
                     type: "Program",
                     line: 1,
                     column: 1

